src/encoder.rs

```diff
@@ -253,67 +253,11 @@ impl<W: Write> Writer<W> {
             ));
         }
 
-        self.w.write_all(&[137, 80, 78, 71, 13, 10, 26, 10])?;
-        let mut data = [0; 13];
-        (&mut data[..]).write_be(self.info.width)?;
-        (&mut data[4..]).write_be(self.info.height)?;
-        data[8] = self.info.bit_depth as u8;
-        data[9] = self.info.color_type as u8;
-        data[12] = if self.info.interlaced { 1 } else { 0 };
-        self.write_chunk(chunk::IHDR, &data)?;
-
-        if let Some(p) = &self.info.palette {
-            write_chunk(&mut self.w, chunk::PLTE, p)?;
-        };
-
-        if let Some(t) = &self.info.trns {
-            write_chunk(&mut self.w, chunk::tRNS, t)?;
-        }
-
-        // If specified, the sRGB information overrides the source gamma and chromaticities.
-        if let Some(srgb) = &self.info.srgb {
-            let gamma = crate::srgb::substitute_gamma().into_scaled().to_be_bytes();
-            let chromaticities =
-                Self::chromaticities_to_be_bytes(&crate::srgb::substitute_chromaticities());
-            write_chunk(&mut self.w, chunk::sRGB, &[srgb.into_raw()])?;
-            write_chunk(&mut self.w, chunk::gAMA, &gamma)?;
-            write_chunk(&mut self.w, chunk::cHRM, &chromaticities)?;
-        } else {
-            if let Some(g) = &self.info.source_gamma {
-                write_chunk(&mut self.w, chunk::gAMA, &g.into_scaled().to_be_bytes())?;
-            }
-
-            if let Some(c) = &self.info.source_chromaticities {
-                let enc = Self::chromaticities_to_be_bytes(&c);
-                write_chunk(&mut self.w, chunk::cHRM, &enc)?;
-            }
-        }
+        self.info.encode(&mut self.w)?;
```

Comment:
Seems like you've accidentally dropped writing the `png` signature itself :joy:

Reply:
Yeah just noticed😅